gui/src/granite/GraniteWizard.tsx

```diff
@@ -699,13 +737,24 @@ const WizardContent: React.FC = () => {
         }
         case "status": {
           const data = payload.data;
-          setServerStatus(data.serverStatus);
+          const newServerState = data.serverState;
+          const newVersion: string | undefined = newServerState.version;
+          const isOllamaOutdated = newVersion !== undefined && !checkMinimumServerVersion(newVersion);
+          setOllamaOutdated(isOllamaOutdated);
+          setServerState((oldState) => {
+            if (oldState.version !== undefined && oldState.version !== newServerState.version && isOllamaOutdated) {
+              setActiveStep(OLLAMA_STEP);
+            }
```

Comment:
Can this be put into setStepStatuses() - if the status of OLLAMA_STEP goes from ready to not-ready, return to the ollama step? Controlling the active step as a side effect of the update function for stepStatuses is clever, controlling it as a side-effect of the update function for the state is ugly. :wink:

Reply:
good call